e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js

```diff
@@ -1212,3 +1212,74 @@ describe("issue 49304", () => {
     });
   });
 });
+
+describe("issue 50925", () => {
+  const questionDetails = {
+    query: {
+      "source-table": PRODUCTS_ID,
+      expressions: {
+        Custom: [
+          "case",
+          [
+            [
+              [
+                "=",
+                ["field", PRODUCTS.ID, { "base-type": "type/BigInteger" }],
+                1,
+              ],
+              [
+                "*",
+                ["field", PRODUCTS.PRICE, { "base-type": "type/Float" }],
+                1.21,
+              ],
+            ],
+          ],
+          {
+            default: ["field", PRODUCTS.PRICE, { "base-type": "type/Float" }],
+          },
+        ],
+      },
+    },
+  };
+
+  beforeEach(() => {
+    H.restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should not remove existing characters when applying autocomplete suggestion (metabase#50925)", () => {
+    H.createQuestion(questionDetails, { visitQuestion: true });
+    H.openNotebook();
+
+    cy.log("incomplete bracket identifier is followed by whitespace");
+    H.getNotebookStep("expression").findByText("Custom").click();
+    cy.get(".ace_text-input")
+      .first()
+      .focus()
+      .type("{leftarrow}".repeat(9))
+      .type(" [Pr{enter}");
+
+    cy.get(".ace_text-input")
+      .first()
+      .should(
+        "have.value",
+        "case([ID] = 1, [Price] * 1.21, [Price]  [Price])\n\n",
+      );
+
+    cy.log("incomplete bracket identifier is followed by bracket identifier");
+    H.popover().button("Cancel").click();
+    H.getNotebookStep("expression").findByText("Custom").click();
+    cy.get(".ace_text-input")
```

Comment:
Please use helpers like `H.focusNativeEditor()`

Reply:
These helpers work only with `NativeQueryEditor` which is not used here :(

`H.enterCustomColumnDetails` also didn't really fit here because it clears the input.